### PR TITLE
My final solution to the ShakespeareAnalyzer problem

### PR DESCRIPTION
**Pull request by charlieanna**

### CODE REVIEW

**Comment by charlieanna** [comment]
Please help me to write more specs. I don't understand the edge cases here.

File: result.txt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+5 Seyton
+5 Servant
+4 Second apparition
+3 Lords
+2 First apparition
+2 Fleance
+2 Both murderers
+1 Attendant
+1 Soldiers
+1 Macbethlennox
```
Comment (qpowell) [inline result.txt:42]: Here is an edge case you can test for. There are speeches that contain more than one speaker. This one caught me off guard as well!

Example:

```
<SPEECH>
<SPEAKER>MACBETH</SPEAKER>
<SPEAKER>LENNOX</SPEAKER>
<LINE>What's the matter.</LINE>
</SPEECH>
```

Hope this helps point you in the right direction :)

Comment (charlieanna) [inline result.txt:42]: Thanks :). Will test it out.

Comment (yangtheman) [inline result.txt:42]: Oh, I didn't catch this either.

File: lib/file_content.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+require 'content'
+require 'nokogiri'
+class FileContent < Content
+  def get_content_from_source
+    f = File.open(@source)
+    Nokogiri::XML(f)
```
Comment (yangtheman) [inline lib/file_content.rb:6]: I think `Nokogiri::XML(open())` works for both opening a local file and reading remote file.

File: lib/content.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+require 'nokogiri'
+class Content
+  attr_accessor :data,:source
+  def initialize(source)
+    @source = source
+    @data = get_content_from_source
+  end
+  
+  def get_content_from_source
+  end
```
Comment (yangtheman) [inline lib/content.rb:10]: I think you can raise an exception here in case this method is called from an instance of Content class directly.
